GitUI/CommandsDialogs/SettingsDialog/Pages/ChecklistSettingsPage.cs

```diff
@@ -502,81 +482,42 @@ private bool CheckGitCmdValid()
+            if (string.IsNullOrEmpty(diffTool))
             {
                 RenderSettingUnset(DiffTool, DiffTool_Fix, _adviceDiffToolConfiguration.Text);
                 return false;
             }
 
-            string cmd = GetGlobalSetting($"difftool.{difftool}.cmd");
+            string cmd = _diffMergeToolConfigurationManager.GetToolCommand(diffTool, DiffMergeToolType.Diff);
             if (cmd.IsNullOrWhiteSpace())
             {
-                cmd = MergeToolsHelper.DiffToolCmdSuggest(difftool, "");
-                if (cmd.IsNullOrWhiteSpace())
-                {
-                    RenderSettingUnset(DiffTool, DiffTool_Fix, _adviceDiffToolConfiguration.Text);
-                    return false;
-                }
+                RenderSettingUnset(DiffTool, DiffTool_Fix, _adviceDiffToolConfiguration.Text);
+                return false;
             }
 
-            RenderSettingSet(DiffTool, DiffTool_Fix, string.Format(_diffToolXConfigured.Text, difftool));
+            RenderSettingSet(DiffTool, DiffTool_Fix, string.Format(_diffToolXConfigured.Text, diffTool));
             return true;
         }
 
         private bool CheckMergeTool()
         {
             MergeTool.Visible = true;
-            string mergetool = CommonLogic.GetGlobalMergeTool();
-            if (string.IsNullOrEmpty(mergetool))
+            var mergeTool = _diffMergeToolConfigurationManager.ConfiguredMergeTool;
+            if (string.IsNullOrEmpty(mergeTool))
             {
                 RenderSettingUnset(MergeTool, MergeTool_Fix, _configureMergeTool.Text);
                 return false;
             }
 
-            // Hardcode parameters to some mergetools
-            if (EnvUtils.RunningOnWindows())
-            {
-                if (mergetool.ToLowerInvariant() == "kdiff3")
-                {
-                    string p = GetGlobalSetting($"mergetool.{mergetool}.path");
-                    if (string.IsNullOrEmpty(p) || !File.Exists(p))
-                    {
-                        RenderSettingUnset(MergeTool, MergeTool_Fix, string.Format(_mergeToolXConfiguredNeedsCmd.Text, mergetool));
-                        return false;
-                    }
-
-                    RenderSettingSet(MergeTool, MergeTool_Fix, string.Format(_customMergeToolXConfigured.Text, mergetool));
-                    return true;
-                }
-
-                if (mergetool.ToLowerInvariant() == "tmerge")
-                {
-                    string p = GetGlobalSetting($"mergetool.{mergetool}.cmd");
-                    if (string.IsNullOrEmpty(p))
-                    {
-                        RenderSettingUnset(MergeTool, MergeTool_Fix, string.Format(_mergeToolXConfiguredNeedsCmd.Text, mergetool));
-                        return false;
-                    }
-
-                    RenderSettingSet(MergeTool, MergeTool_Fix, string.Format(_customMergeToolXConfigured.Text, mergetool));
-                    return true;
-                }
```

Comment:
This special handling is now removed. Probably OK

Reply:
I don't want any more special handling, it makes the code incoherent mess.

Reply:
Neither do I, but the behavior changes, this at least must be documented in the commit message. It is very hard to investigate changes otherwise.

Reply:
Done

Reply:
pushed?